modules/helm-adding-namespace-scoped-helm-chart-repositories.adoc

```diff
@@ -0,0 +1,92 @@
+
+[NOTE]
+====
+* Administrators can limit users from creating namespace-scoped Helm repository resources. By limiting users, admins have the flexibility to control the role-based access control (RBAC) through a namespace role instead of a cluster role. Doing so avoids unnecessary permission elevation for the user and prevents access to unauthorized services or applications.
+* The addition of the namespace-scoped Helm repository does not impact the behavior of the existing cluster-scoped Helm repository.
+====
+
+As a regular user or project member with the appropriate RBAC permissions, you can add custom Helm chart repositories to your cluster and enable access to the Helm charts from these repositories in the *Developer Catalog*.
+
+.Procedure
+
+. To add a new namespace-scoped Helm Chart Repository, you must add the Helm Chart Repository custom resource (CR) to your namespace.
++
+.Sample Namespace-scoped Helm Chart Repository CR
+
+[source,yaml]
+----
+apiVersion: helm.openshift.io/v1beta1
+kind: ProjectHelmChartRepository
+metadata:
+  name: <name>
+spec:
+  url: https://my.chart-repo.org/stable
+
+  # optional name that might be used by console
+  name: <chart-repo-display-name>
+
+  # optional and only needed for UI purposes
+  description: <My private chart repo>
+
+  # required: chart repository URL
+  connectionConfig:
+    url: <helm-chart-repository-url>
+----
++
+For example, to add an Azure sample chart repository scoped to your `my-namespace` namespace, run:
++
+[source,terminal]
+----
+$ cat <<EOF | oc apply --namespace my-namespace -f -
+apiVersion: helm.openshift.io/v1beta1
+kind: ProjectHelmChartRepository
+metadata:
+  name: azure-sample-repo
+spec:
+  name: azure-sample-repo
+  connectionConfig:
+    url: https://raw.githubusercontent.com/Azure-Samples/helm-charts/master/docs
+EOF
+----
++
+The output verifies that the namespace-scoped Helm Chart Repository CR is created:
++
+.Example output
+----
+projecthelmchartrepository.helm.openshift.io/azure-sample-repo created
+----
+
+. Navigate to  the *Developer Catalog* in the web console to verify that the Helm charts from the chart repository are displayed in your `my-namespace` namespace.
++
+For example, use the *Chart repositories* filter to search for a Helm chart from the repository.
++
+.Chart repositories filter in your namespace
+image::odc_namespace_helm_chart_repo_filter.png[]
++
+Alternatively, run:
++
+[source,terminal]
+----
+$ oc get projecthelmchartrepositories --namespace my-namespace
+----
++
+.Example output
+----
+NAME          AGE
+azure-sample-repo        1m
+----
++
+[NOTE]
+====
+If a cluster administrator removes all of the chart repositories, then you cannot view the Helm option in the *+Add* view, *Developer Catalog*, and left navigation panel.
+====
```

Comment:
@zonggen @dperaza4dustbit: Does this hold true even when a regular user/project member with appropriate RBAC permissions removes all of the chart repositories? 

In that case, should the Note say: "If a cluster administrator or a regular user with appropriate RBAC permissions removes all of the chart repositories, then you cannot view the Helm option in the *+Add* view, *Developer Catalog*, and left navigation panel."? Please clarify. Thanks!

Reply:
Yes, but a user can add a repository on their namespace and it will be enabled for them.

Reply:
The admin would have the option to remove permissions for a user under a project to be able to create ProjectHelmChartRepository resources.

Reply:
> Yes, but a user can add a repository on their namespace and it will be enabled for them.

Thanks, @dperaza4dustbit! Question: what is enabled for them? The functionality to add a repo? Or the repo itself (if yes, then what is it enabled for)? Please clarlify.

Reply:
> The admin would have the option to remove permissions for a user under a project to be able to create ProjectHelmChartRepository resources.

Thanks, @dperaza4dustbit! I think we have mentioned this as a Note at the beginning of the adding namespace-scoped repo procedure: "Administrators can limit users from creating namespace-scoped Helm repository resources."
Question: 
- Is your input different from what is mentioned in the note section above? If yes, then please explain the difference and clarify.
  - In that case, is this worth mentioning in the note section here?

Reply:
> Thanks, @dperaza4dustbit! Question:
> 
> * what is enabled for them? The functionality to add a repo? Or the repo itself (if yes, then what is it enabled for)? Please clarify.
> * Is this worth mentioning in the note section here? WDYT?

Prior to Namespace scope repos the admin could disable our default repo and the helm feature would be completely disabled. Now that each project can create repos at the namespace level, users in that specific project the has a repo will be able to see the helm features, while users in another project without repos will not see the helm feature cause. So I think this is worth mentioning cause the admin disabling default cluster scope repo is not enough any more to disable helm. Each project has now the ability to re-enable helm from them. See the console demo recording showing some of this: https://drive.google.com/file/d/1dKuzf8f9pVTm_v_J4xgA7l6EsW4jQqfJ/view?usp=sharing

Reply:
> Thanks, @dperaza4dustbit! I think we have mentioned this as a Note at the beginning of the adding namespace-scoped repo procedure: "Administrators can limit users from creating namespace-scoped Helm repository resources." Question:
> 
> * Is your input different from what is mentioned in the note section above? If yes, then please explain the difference and clarify.
>   
>   * In that case, is this worth mentioning in the note section here?

This is good, yes

Reply:
@dperaza4dustbit, @zonggen: Modifying the Note section content like this: "If a cluster administrator or a regular user with appropriate RBAC permissions removes all of the chart repositories in a specific namespace, then you cannot view the Helm option in the +Add view, Developer Catalog, and left navigation panel for that specific namespace."

PTAL and let me know.

Reply:
I think the modification is correct ^^